_episodes/10-zonal-statistics.md

```diff
@@ -0,0 +1,188 @@
+4868	Grasland	Grasland, blijvend	265	2020	Definitief	POLYGON ((635240.367 5806904.896, 635245.819 5...
+3061 rows × 6 columns
+~~~
+{: .output}
+
+# Rasterizing our vector data
+
+Before calculating zonal statistics, we first need to rasterize our `field_cropped` vector geodataframe with the `rasterio.features.rasterize` function. With this function, we aim to produce a grid with numerical values representing the types of crop as defined by the column `gewascode` from `field_cropped` - `gewascode` stands for the crop codes as defined by the Netherlands Enterprise Agency (RVO) for different types of crops or `gewas` (Grassland, permanent; Grassland, temporary; corn fields; etc.). This grid of values thus defines the zones for the `xrspatial.zonal_stats` function, where each pixel in the zone grid overlaps with a corresponding pixel in our NDVI raster. 
+
+We can generate the `geometry, gewascode` pairs for each vector data-point to be used as the first argument to `rasterio.features.rasterize` as:
+
+~~~
+geom = field_cropped[['geometry', 'gewascode']].values.tolist()
+geom
+~~~
+{: .language-python}
+
+~~~
+[[<shapely.geometry.polygon.Polygon at 0x7ff88666f670>, 265],
+ [<shapely.geometry.polygon.Polygon at 0x7ff86bf39280>, 265],
+ [<shapely.geometry.polygon.Polygon at 0x7ff86ba1db80>, 265],
+ [<shapely.geometry.polygon.Polygon at 0x7ff86ba1d730>, 265],
+ [<shapely.geometry.polygon.Polygon at 0x7ff86ba1d400>, 265],
+ [<shapely.geometry.polygon.Polygon at 0x7ff86ba1d130>, 265],
+...
+ [<shapely.geometry.polygon.Polygon at 0x7ff88685c970>, 265],
+ [<shapely.geometry.polygon.Polygon at 0x7ff88685c9a0>, 265],
+ [<shapely.geometry.polygon.Polygon at 0x7ff88685c9d0>, 265],
+ [<shapely.geometry.polygon.Polygon at 0x7ff88685ca00>, 331],
+ ...]
+~~~
+{: .output}
+
+This generates a list of the shapely geometries from the `geometry` column, and the unique field ID from the `gewascode` column in the `fields_cropped` geodataframe.
+
+We can now rasterize our vector data using `rasterio.features.rasterize`:
+
+~~~
+from rasterio import features 
+field_cropped_raster = rasterio.features.rasterize(geom, out_shape=ndvi_sq.shape, fill=0, transform=ndvi.rio.transform())
+~~~
+{: .language-python}
+
+The argument `out_shape` specifies the shape of the output grid in pixel units, while `transform` represents the projection from pixel space to the projected coordinate space. We also need to specify the fill value for pixels that are not contained within a polygon in our shapefile, which we do with `fill = 0`. It's important to pick a fill value that is not the same as any value already defined in `gewascode` or else we won't distinguish between this zone and the background.
+
+We convert the output of the `rasterio.features.rasterize` function, which generates a numpy array `np.ndarray`, to `xarray.DataArray` which will be used further:
+~~~
+import xarray as xr
+field_cropped_raster_xarr = xr.DataArray(field_cropped_raster)
```

Comment:
This is actually `cropfield_raster_xarr` below - correct?

Reply:
Yes it is. Fixed now.